collect_app/src/androidTest/java/org/odk/collect/android/feature/instancemanagement/FormResubmissionTest.java

```diff
@@ -0,0 +1,151 @@
+package org.odk.collect.android.feature.instancemanagement;
+
+import android.Manifest;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.GrantPermissionRule;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.RecordedIntentsRule;
+import org.odk.collect.android.dao.CursorLoaderFactory;
+import org.odk.collect.android.openrosa.HttpCredentialsInterface;
+import org.odk.collect.android.openrosa.HttpPostResult;
+import org.odk.collect.android.support.CollectTestRule;
+import org.odk.collect.android.support.StubOpenRosaServer;
+import org.odk.collect.android.support.TestDependencies;
+import org.odk.collect.android.support.TestRuleChain;
+import org.odk.collect.android.support.pages.MainMenuPage;
+import org.odk.collect.android.support.pages.SendFinalizedFormPage;
+
+import java.io.File;
+import java.net.URI;
+import java.util.List;
+
+import timber.log.Timber;
+
+@RunWith(AndroidJUnit4.class)
+public class FormResubmissionTest {
+
+    public static final String TEXT_FORM_NAME = "One Question";
+    public static final String TEXT_FORM_XML = "one-question.xml";
+    public static final String TEXT_QUESTION = "what is your age";
+    public static final String TEXT_ANSWER = "123";
+
+    private boolean noHttpPostResult;
+    private boolean rejectResubmission;
+    private File submissionFile;
+
+    private final StubOpenRosaServer server = new StubOpenRosaServer() {
+        @NonNull
+        @Override
+        public HttpPostResult uploadSubmissionAndFiles(@NonNull File submissionFile,
+                                                       @NonNull List<File> fileList,
+                                                       @NonNull URI uri,
+                                                       @Nullable HttpCredentialsInterface credentials,
+                                                       long contentLength) throws Exception {
+            if (noHttpPostResult || rejectResubmission) {
+                boolean doReject = handleFlag(submissionFile);
+                if (doReject) {
+                    return new HttpPostResult("", 500, "Resubmission not permitted for " + submissionFile.getName());
+                }
+            }
+
+            return super.uploadSubmissionAndFiles(submissionFile,
+                    fileList,
+                    uri,
+                    credentials,
+                    contentLength);
+        }
+    };
+
+    private boolean handleFlag(@NotNull File submissionFile) throws InterruptedException {
```

Comment:
I don't think any of this is needed. You just need any error state, not necessarily a timeout. See https://github.com/getodk/collect/issues/4589#issuecomment-857008731 for some ideas.